Upgrade to a proper async worker for sync call, fixes #138

kriszyp opened the pull request.

Venemo commented on src/env.cpp:
```diff
@@ -98,6 +98,36 @@ int applyUint32Setting(int (*f)(MDB_env *, T), MDB_env* e, Local<Object> options
+
+/*    ~SyncWorker() {
+      delete env;
+    }*/
```
You deleted the same lines in your previous commit, I assume these are here by mistake?

kriszyp replied:
I think I deleted it in the later/2nd commit. It makes sense that isn't needed, right?

Venemo commented on src/env.cpp:
```diff
@@ -307,37 +337,16 @@ NAN_METHOD(EnvWrap::sync) {
 
-    Local<Function> callback = info[0].As<Function>();
-
-    EnvSyncData *d = new EnvSyncData;
```
This was the only place where `EnvSyncData` was ever used, can you please delete it now that it's not used anymore?

kriszyp commented:
I added the unit test for this issue (which fails before fix and passes afterwards for me), and cleaned up EnvSyncData. Let me know if there is anything else you want me to do. Thanks!

Venemo commented:
Thanks. Merged now.
